### PR TITLE
Updated HMRC mandatory facts

### PR DESCRIPTION
**Pull request by allenbross-wf**
#### Description of change

Updated HMRC mandatory facts validation

Added HMRC conformance suite

#### Steps to Test

CI

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: arelle/plugin/validate/HMRC/__init__.py [hunk 1]
```diff
@@ -265,6 +272,8 @@ def checkFacts(facts):
                     factLocalName = f.qname.localName
                     if factLocalName in mandatoryItems[val.txmyType]:
                         mandatoryFacts[factLocalName] = f
+                    if val.txmyType == "charities" and factLocalName in mustHaveOneItem["charities"]:
```
Comment (aaroncameron-wk) [inline arelle/plugin/validate/HMRC/__init__.py:275]: Could make this more generic with
```suggestion
                    if val.txmyType in mustHaveOneItem and factLocalName in mustHaveOneItem[val.txmyType]:
```

File: arelle/plugin/validate/HMRC/__init__.py [hunk 2]
```diff
@@ -345,6 +354,10 @@ def checkFacts(facts):
                 modelXbrl.error("JFCVC.3312",
                     _("Facts are MANDATORY: %(missingItems)s"),
                     modelObject=modelXbrl, missingItems=", ".join(sorted(_missingItems)))
+            if not atLeastOneFacts and val.txmyType == "charities":
+                modelXbrl.error("JFCVC.3312.char",
+                                _("At least one of the facts is MANDATORY: %(missingItems)s"),
+                                modelObject=modelXbrl, missingItems=", ".join(sorted(mustHaveOneItem["charities"])))
```
Comment (aaroncameron-wk) [inline arelle/plugin/validate/HMRC/__init__.py:360]: Similar to other suggestion, we can avoid hardcoding `"charities"` here and just use `val.txmyType in mustHaveOneItem`

File: arelle/plugin/validate/HMRC/__init__.py [hunk 3]
```diff
@@ -29,7 +29,7 @@
     "EntityCurrentLegalOrRegisteredName", "StartDateForPeriodCoveredByReport",
     "EndDateForPeriodCoveredByReport", "BalanceSheetDate"}
 
-COMMON_MANDATORY_FRS_ITEMS = commonMandatoryItems | {
+commonMandatoryFrsItems = commonMandatoryItems | {
```
Comment (derekgengenbacher-wf) [inline arelle/plugin/validate/HMRC/__init__.py:32]: I believe that common python formatting is actually capitalized snake case for constants such as these. So `COMMON_MANDATORY_FRS_ITEMS` should be the name. Anything that doesnt match that is what actually needs to be updated. Further below, mustHaveOneItem should be `MUST_HAVE_ONE_ITEM`.

File: arelle/plugin/validate/HMRC/__init__.py [hunk 4]
```diff
@@ -43,12 +43,19 @@
     "charities": commonMandatoryItems | {
         "DateAuthorisationFinancialStatementsForIssue", "DirectorSigningFinancialStatements",
         "EntityDormantTruefalse", "EntityTradingStatus",
-        "AccountingStandardsApplied", "AccountingStandardsApplied"},
+        "AccountingStandardsApplied", "AccountsStatusAuditedOrUnaudited"},
```
Comment (brettkail-wk) [inline arelle/plugin/validate/HMRC/__init__.py:46]: Was this always wrong, or does this change only apply to the newer taxonomy?  Should we instead check `val.modelXbrl.nameConcepts.get("AccountsStatusAuditedOrUnaudited", ())` and add a `charities-2023` key instead?

Comment (allenbross-wf) [reply arelle/plugin/validate/HMRC/__init__.py:46]: This seems to have always been wrong.  I can't find a version of the Joint Filing Common Validation Checks prior to v4. In V4 both  "AccountingStandardsApplied" and "AccountsStatusAuditedOrUnaudited" have required for the charity taxonomies from 2015 to present.

File: arelle/plugin/validate/HMRC/__init__.py [hunk 5]
```diff
@@ -25,32 +25,39 @@
 EMPTYDICT = {}
 _6_APR_2008 = dateTime("2008-04-06", type=DATE)
 
-commonMandatoryItems = {
+CommonMandatoryItems = {
```
Comment (derekgengenbacher-wf) [inline arelle/plugin/validate/HMRC/__init__.py:28]: nit: This should be all caps and snake case

**Review by derekgengenbacher-wf — DISMISSED** [review]

**Review by derekgengenbacher-wf — APPROVED** [review]

**Review by aaroncameron-wk — APPROVED** [review]

**Review by austinmatherne-wk — APPROVED** [review]

**Pull request merged**